Sources/Rosalind/Rosalind.swift

```diff
@@ -1 +1,88 @@
+import Command
+import CryptoKit
+import FileSystem
+import Foundation
+import Path
 
+enum RosalindError: LocalizedError {
+    case notFound(AbsolutePath)
+
+    var errorDescription: String? {
+        switch self {
+        case let .notFound(path):
+            return "File not found at path \(path.pathString)"
+        }
+    }
+}
+
+public protocol Rosalindable {
+    func analyze(path: AbsolutePath) async throws -> Report
+}
+
+public struct Rosalind: Rosalindable {
+    let fileSystem: FileSysteming
+    let commandRunner: CommandRunning
+
+    public init() {
+        self.init(fileSystem: FileSystem(), commandRunner: CommandRunner())
+    }
+
+    init(
+        fileSystem: FileSysteming,
+        commandRunner: CommandRunning
+    ) {
+        self.fileSystem = fileSystem
+        self.commandRunner = commandRunner
+    }
+
+    public func analyze(path: AbsolutePath) async throws -> Report {
```

Comment:
nit: `Report` feels like a quite generic name for something part of the public API. I'd consider naming it something like `RosalindReport`, `RosalindReportTree` `ReportTree` or somehow make it a bit more concrete.